sim-cli/src/parsing.rs

```diff
@@ -0,0 +1,374 @@
+    CLN(cln::ClnConnection),
+}
+
+/// Data structure used to parse information from the simulation file. It allows source and destination to be
+/// [NodeId], which enables the use of public keys and aliases in the simulation description.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ActivityParser {
+    /// The source of the payment.
+    #[serde(with = "serializers::serde_node_id")]
+    pub source: NodeId,
+    /// The destination of the payment.
+    #[serde(with = "serializers::serde_node_id")]
+    pub destination: NodeId,
+    /// The time in the simulation to start the payment.
+    pub start_secs: Option<u16>,
+    /// The number of payments to send over the course of the simulation.
+    #[serde(default)]
+    pub count: Option<u64>,
+    /// The interval of the event, as in every how many seconds the payment is performed.
+    #[serde(with = "serializers::serde_value_or_range")]
+    pub interval_secs: Interval,
+    /// The amount of m_sat to used in this payment.
+    #[serde(with = "serializers::serde_value_or_range")]
+    pub amount_msat: Amount,
+}
+
+impl TryFrom<&Cli> for SimulationCfg {
+    type Error = anyhow::Error;
+
+    fn try_from(cli: &Cli) -> Result<Self, Self::Error> {
+        Ok(SimulationCfg::new(
+            cli.total_time,
+            cli.expected_pmt_amt,
+            cli.capacity_multiplier,
+            if !cli.no_results {
+                Some(WriteResults {
+                    results_dir: mkdir(cli.data_dir.join("results"))?,
+                    batch_size: cli.print_batch_size,
+                })
+            } else {
+                None
+            },
+            cli.fix_seed,
+        ))
+    }
+}
+
+/// Parses the cli options provided and creates a simulation to be run, connecting to lightning nodes and validating
+/// any activity described in the simulation file.
+pub async fn create_simulation(
+    cli: &Cli,
+) -> Result<Simulation, anyhow::Error> {
+
+    let cfg: SimulationCfg = SimulationCfg::try_from(cli)?;
+
+    let sim_path = read_sim_path(cli.data_dir.clone(), cli.sim_file.clone()).await?;
+    let SimParams { nodes, activity } = serde_json::from_str(&std::fs::read_to_string(sim_path)?)
+        .map_err(|e| {
+        anyhow!(
+            "Could not deserialize node connection data or activity description from simulation file (line {}, col {}, err: {}).",
+            e.line(),
+            e.column(),
+            e.to_string()
+        )
+    })?;
+
+    let (clients, clients_info) = get_clients(nodes).await?;
+    // We need to be able to look up destination nodes in the graph, because we allow defined activities to send to
+    // nodes that we do not control. To do this, we can just grab the first node in our map and perform the lookup.
+    // block_on is used to avoid asynchronous closures, which is okay because we expect these lookups to be very cheap.
+    let get_node = |pk: &PublicKey| -> Result<NodeInfo, LightningError> {
```

Comment:
since async blocks on rust are already stable, we could use them here. https://github.com/rust-lang/rust/pull/132706 and not use `block_on`